auth_token/utils.py

```diff
@@ -176,21 +306,271 @@ def get_user(request):
+
+    Returns:
+        True if OTP expiration was extended, False otherwise
+    """
+    otp = get_valid_otp(slug, key)
+    if otp is None:
+        return False
+    else:
+        otp.change_and_save(expires_at=compute_expires_at(expiration or settings.DEFAULT_OTP_AGE))
+        return True
+
+
+def create_authorization_request(type, slug, user, title, description=None, authorization_token=None, related_objects=None,
+                                 data=None, otp_key_generator=None, otp_sender=None, mobile_device=None,
+                                 expiration=None):
     """
-    Random ID generating of 'length' bytes
+    Create authorization request with a given type slug, title and description.
+    Args:
+        type: authorization request type OTP or MOBILE_DEVICE.
+        slug: string for authorization request identification.
+        user: owner of authorization request.
+        title: human readable title for a client.
+        description: human readable description for a client.
+        authorization_token: authorization request can be valid only for specific authorization token.
+        related_objects: related model instances which will be stored with authorization request.
+        data: extra authorization request data.
+        otp_key_generator: OTP key generator. Only used for OTP type.
+        otp_sender: OTP code sender. Only used for OTP type.
+        mobile_device: mobile device which can grant/deny authorization. Only used for DEVICE type.
+        expiration: expiration time in seconds.
+
+    Returns:
+        new authorization token.
     """
-    return force_text(binascii.hexlify(os.urandom(length)))
+    from .models import AuthorizationRequest
+
+    expiration = expiration or settings.DEFAULT_AUTHORIZATION_REQUEST_AGE
+
+    if type == AuthorizationRequestType.OTP:
+        otp = create_otp(slug, key_generator=otp_key_generator, expiration=expiration)
+        authorization_request = AuthorizationRequest.objects.create(
+            slug=slug,
+            user=user,
+            title=title,
+            description=description,
+            type=type,
+            one_time_password=otp,
+            authorization_token=authorization_token,
+            data=data,
+            expires_at=compute_expires_at(expiration)
+        )
+        if related_objects:
+            authorization_request.related_objects.add(*related_objects)
+        if otp_sender:
+            otp_sender(authorization_request, otp.secret_key)
+    else:
+        authorization_request = AuthorizationRequest.objects.create(
+            slug=slug,
+            user=user,
+            title=title,
+            description=description,
+            type=type,
+            mobile_device=mobile_device,
+            authorization_token=authorization_token,
+            data=data,
+            expires_at=compute_expires_at(expiration)
+        )
```

Comment:
Very similar kwargs duplicated twice, put them into a variable and just do update/add the differing paramter.

Reply:
this was changed